drivers/sensor/bh1745/bh1745.c

```diff
@@ -0,0 +1,312 @@
+		  BH1745_MODE_CONTROL2_RGB_EN_DISABLE;
+
+	return i2c_reg_update_byte(data->i2c,
+				   DT_REG_ADDR(DT_DRV_INST(0)),
+				   BH1745_MODE_CONTROL2,
+				   BH1745_MODE_CONTROL2_RGB_EN_Msk,
+				   en);
+}
+
+
+static void bh1745_async_init(struct k_work *work)
+{
+	struct bh1745_data *data = bh1745_dev->data;
+
+	ARG_UNUSED(work);
+
+	LOG_DBG("BH1745 async init step %d", data->async_init_step);
+
+	data->err = async_init_fn[data->async_init_step](data);
+
+	if (data->err) {
+		LOG_ERR("BH1745 initialization failed");
+	} else {
+		data->async_init_step++;
+
+		if (data->async_init_step == ASYNC_INIT_STEP_COUNT) {
+			data->ready = true;
+			LOG_INF("BH1745 initialized");
+		} else {
+			k_work_schedule(&bh1745_init_work,
+					K_MSEC(async_init_delay[data->async_init_step]));
+		}
+	}
+}
+
+static int bh1745_async_init_reset_check(struct bh1745_data *data)
+{
+	(void)memset(data->sample_rgb_light, 0, sizeof(data->sample_rgb_light));
+	int err;
+
+	err = bh1745_sw_reset(data->i2c);
+
+	if (err) {
+		LOG_ERR("Could not apply software reset.");
+		return -EIO;
+	}
+	err = bh1745_check(data);
+	if (err) {
+		LOG_ERR("Communication with BH1745 failed with error %d", err);
+		return -EIO;
+	}
+	return 0;
+}
+
+static int bh1745_async_init_rgb_enable(struct bh1745_data *data)
+{
+	int err;
+
+	err = bh1745_rgb_measurement_enable(data, true);
+	if (err) {
+		LOG_ERR("Could not set measurement mode.");
+		return -EIO;
+	}
+	return 0;
+}
+
+static int bh1745_async_init_configure(struct bh1745_data *data)
+{
+	if (i2c_reg_write_byte(data->i2c, DT_REG_ADDR(DT_DRV_INST(0)),
+			       BH1745_MODE_CONTROL1,
+			       BH1745_MODE_CONTROL1_DEFAULTS)) {
+		LOG_ERR(
+			"Could not set gain and measurement mode configuration.");
+		return -EIO;
+	}
+
+#ifdef CONFIG_BH1745_TRIGGER
+	int err = bh1745_gpio_interrupt_init(bh1745_dev);
+	if (err) {
+		LOG_ERR("Failed to initialize interrupt with error %d", err);
+		return -EIO;
+	}
+
+	LOG_DBG("GPIO Sense Interrupts initialized");
+#endif /* CONFIG_BH1745_TRIGGER */
+
+	return 0;
+}
+
+static int bh1745_init(const struct device *dev)
+{
+	bh1745_dev = dev;
+
+	struct bh1745_data *data = bh1745_dev->data;
+	data->i2c = device_get_binding(DT_BUS_LABEL(DT_DRV_INST(0)));
+
+	if (data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!",
+			DT_BUS_LABEL(DT_DRV_INST(0)));
+
+		return -EINVAL;
+	}
+	k_work_init_delayable(&bh1745_init_work, bh1745_async_init);
+	k_work_schedule(&bh1745_init_work, K_MSEC(async_init_delay[data->async_init_step]));
+	return 0;
+};
+
+static const struct sensor_driver_api bh1745_driver_api = {
+	.sample_fetch = &bh1745_sample_fetch,
+	.channel_get = &bh1745_channel_get,
+#ifdef CONFIG_BH1745_TRIGGER
+	.attr_set = &bh1745_attr_set,
+	.trigger_set = &bh1745_trigger_set,
+#endif
+};
+
+static struct bh1745_data bh1745_data;
+
+DEVICE_DEFINE(bh1745, DT_INST_LABEL(0),
+	      bh1745_init, NULL, &bh1745_data, NULL,
+	      POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &bh1745_driver_api);
```

Comment:
Please update this driver to support multiple instances and use `DT_INST_FOREACH_STATUS_OKAY`

Reply:
Done